src/crc32c.c

```diff
@@ -114,14 +109,15 @@ static uint32_t crc32c_sse42(uint32_t crc, const char *buf, size_t len)
         uintptr_t crc2 = 0;
         const char *end = buf + LONG;
         do {
-            __asm__(CRC32_PTR "\t" "(%3), %0\n\t"
-                    CRC32_PTR "\t" LONGx1 "(%3), %1\n\t"
-                    CRC32_PTR "\t" LONGx2 "(%3), %2"
-                    : "+r"(crc0), "+r"(crc1), "+r"(crc2)
-                    : "r"(buf),
-                      "m"(* (const char (*)[sizeof(void*)]) &buf[0]),
-                      "m"(* (const char (*)[sizeof(void*)]) &buf[LONG]),
-                      "m"(* (const char (*)[sizeof(void*)]) &buf[LONG*2]));
+            __asm__(CRC32_PTR "\t(%1), %0"
+                    : "+r"(crc0)
+                    : "rm"(* (const char (*)[sizeof(void*)]) &buf[0]));
+            __asm__(CRC32_PTR "\t(%1), %0"
+                    :  "+r"(crc1)
+                    : "rm"(* (const char (*)[sizeof(void*)]) &buf[LONG]));
```

Comment:
the generated code seems sort of bad here, but maybe it is because ASAN has no optimizations so it is making a copy of the data onto the stack?
```
<inline asm>:1:14: note: instantiated into assembly here
        crc32q  (-136(%rbp)), %rax
```